components/org.wso2.config.diff.creater/src/main/java/org/wso2/config/diff/creater/ConfigDiffChecker.java

```diff
@@ -0,0 +1,339 @@
+                configLoader.getJ2TemplateFiles(), existingTags, outputGenerator.getOutputCSV(),
+                outputGenerator.getOutputKeyValues());
+    }
+
+    /**
+     * Check whether property files are in templates or not, if true find diff.
+     *
+     * @param defaultPropertiesFiles  default property files map.
+     * @param migratedPropertiesFiles migrated property files map.
+     * @param j2TemplateFiles         template property files .
+     * @param existingTags            map of existing knowledge.
+     * @param outputCSVFile           output csv file.
+     * @param keyValueFile            output key-value file.
+     * @throws IOException IOException
+     */
+    private void checkPropertyDiff(Map<String, File> defaultPropertiesFiles,
+                                   Map<String, File> migratedPropertiesFiles, Map<String, File> j2TemplateFiles,
+                                   Map<String, String> existingTags, File outputCSVFile, File keyValueFile)
+            throws IOException {
+
+        for (Map.Entry<String, File> entry : migratedPropertiesFiles.entrySet()) {
+
+            if (isFileTemplated(defaultPropertiesFiles, j2TemplateFiles, entry)) {
+
+                Set<Map.Entry<String, String>> changedPropertyDiffSet =
+                        findDiffPropertiesFiles(defaultPropertiesFiles.get(entry.getKey()),
+                                migratedPropertiesFiles.get(entry.getKey()));
+                writePropertiesDiffToCSV(migratedPropertiesFiles.get(entry.getKey()), existingTags,
+                        Paths.get(outputCSVFile.getPath()), Paths.get(keyValueFile.getPath()), changedPropertyDiffSet);
+            } else {
+
+                log.warn(entry.getValue().getPath() + " is not templated with toml. \n");
+                File outFile =
+                        new File(MigrationConstants.UN_TEMPLATE_FILE_FOLDER + MigrationConstants
+                                .FILE_SEPARATOR + entry.getKey());
+                FileUtils.copyFile(entry.getValue(), outFile);
+            }
+        }
+    }
+
+
+    /**
+     * Find the difference of property files.
+     *
+     * @param defaultFile  default file to compare with.
+     * @param migratedFile migrated file used to compare.
+     * @return A set of difference.
+     * @throws IOException IOException.
+     */
+    private Set<Map.Entry<String, String>> findDiffPropertiesFiles(File defaultFile, File migratedFile)
+            throws IOException {
+
+        Properties defaultProperties = new Properties();
+        Properties changedProperties = new Properties();
+        try (FileInputStream defaultInputStream = new FileInputStream(defaultFile);
+             FileInputStream changedInputStream = new FileInputStream(migratedFile)) {
+
+            defaultProperties.load(defaultInputStream);
+            changedProperties.load(changedInputStream);
+            Map<String, String> defaultMap = new HashMap<String, String>((Map) defaultProperties);
```

Comment:
defaultMap >> defaultPropertyMap
changedMap >> changedPropertyMap

Reply:
Should be able to use the diamond operator without param types in `HashMap<String, String>`. Also is casting properties to a Map is allowed?

Reply:
Property is an implementation of Map, hence we can cast properties to a Map.

Reply:
Property is an implementation of Map, Hence we can.